src/frontends/tensorflow_common/src/op/reverse_sequence.cpp

```diff
@@ -14,13 +16,35 @@ namespace frontend {
 namespace tensorflow {
 namespace op {
 OutputVector translate_reverse_sequence_op(const NodeContext& node) {
-    default_op_checks(node, 2, {"ReverseSequence"});
+    default_op_checks(node, 2, {"ReverseSequence"}, true);
     auto input = node.get_input(0);
     auto seq_lengths = node.get_input(1);
 
     // retrieve attributes
     auto seq_dim = node.get_attribute<int64_t>("seq_dim");
     auto batch_dim = node.get_attribute<int64_t>("batch_dim", 0);
+    // Handling negative values by adding rank if needed
+    if (seq_dim < 0) {
+        seq_dim += input.get_shape().size();
+    }
+    if (batch_dim < 0) {
+        batch_dim += input.get_shape().size();
+    }
```

Comment:
Not correct because seq_dim is of int64_t type and you pass it into Graph operation Add. You should update it using Select and Rank operations

Reply:
@rkazants cool I'll make the changes. Thanks for input